src/useObserver.ts

```diff
@@ -34,7 +35,15 @@ export function useObserver<T>(fn: () => T, baseComponentName = "observed"): T {
         }
     })
     if (exception) {
+        reaction.current.dispose()
```

Comment:
Isn't it already disposed on unmount? I am not sure if throwing counts as unmount, but it should.

Reply:
Unmount is not triggered if it happened during the first render; as unmount is setup by `useEffect`, so it is not guaranteed to run if the component never mounted. However, the tracking (and hence establishing of observers) does happen in the first render.

Reply:
(I realized this after the whole concurrent mode discussion)

Reply:
I assume the `Reaction` is fine with double calling `dispose` (if that happens)?

Reply:
Double dispose is no issue btw